### ISSUE TITLE
unauthorized: authentication required error when attempting to push image

### ISSUE THREAD

**Issue by kcturner123** [docker/for-win#108]
I have been working through the tutorial for Docker for Windows and everything has worked perfectly until I tried to push the docker-whale image to my newly created repository. I found a number of older issues similar to this for Linux and Mac, but not for the Windows version.

### Actual behavior
C:\Users\Kim\mydockerbuild>docker push kimdockertutorial/docker-whale
The push refers to a repository [docker.io/kimdockertutorial/docker-whale]
e4ab3a6c9931: Preparing
5f70bf18a086: Preparing
d061ee1340ec: Preparing
d511ed9e12e1: Preparing
091abc5148e4: Preparing
b26122d57afa: Waiting
37ee47034d9b: Waiting
528c8710fd95: Waiting
1154ba695078: Waiting
unauthorized: authentication required
### Information
Diagnostic ID: 0932D0A0-6E49-41BA-8AC1-EF2AF558FE63/2016-09-27_12-55-56
![image](https://cloud.githubusercontent.com/assets/13319081/18884292/c82029fe-84b4-11e6-9dd7-7f2714b36329.png)

**Comment by dgageot** [comment]
Hi @kcturner123 thank you for your report. I believe you need to login to the  hub before you push: https://docs.docker.com/engine/getstarted/step_six/

**Comment by kcturner123** [comment]
David,

 

Thanks for the response, but I had already logged in successfully before trying the push. The problem turned out to be that I didn’t understand the repository name and image tagging dependency. The instructions in the tutorial do not really explain it, but I found the information I needed in a different guide. I didn’t understand that the tag for the image had to be username/repository name. Once I properly tagged the image and used the correct parameters for the push, it was successful.

 

Cheers,

 

Kim

**Comment by dgageot** [comment]
@kcturner123 Happy that you managed to find a solution! Closing

**Comment by friism** [comment]
@kcturner123 Is this the guide that you were following? https://docs.docker.com/engine/getstarted/step_six/

**Comment by kcturner123** [comment]
Yes, that was the tutorial that I was following. It wasn’t quite clear to me in that tutorial that I had to name the repository docker-whale for it to work. I found another guide on the web site that explained it in a different way that made it clear to me. I think a little more explanation in the tutorial of how repositories and tags are used or perhaps some links to more detailed explanations, would be helpful.
